src/pow.cpp

```diff
@@ -355,12 +355,16 @@ void multinomial_coefficients_mpz(int m, int n, map_vec_mpz &r)
 RCP<const Basic> pow_expand(const RCP<const Pow> &self)
 {
     RCP<const Basic> _base = expand(self->base_);
+    bool negative_pow = false;
     if (! is_a<Integer>(*self->exp_) || ! is_a<Add>(*_base))
         return self;
```

Comment:
The problem seems to be that the Mul and Pow expressions are not being canonicalised properly. So the is_canonical assertions are failing. I'm getting similar errors in a lot of expressions, like `(1/((x+y)**2 + x)).expand()` , `(1/(sqrt(3)*sqrt(2) + sqrt(6)*I)**4).expand()`, `(1/((x+y)**2 - x**2 - y**2)).expand()`
I'll look into it further after my classes.